internal/venice-common/src/main/java/com/linkedin/venice/blobtransfer/DvcBlobFinder.java

```diff
@@ -0,0 +1,74 @@
+package com.linkedin.venice.blobtransfer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.venice.client.store.transport.TransportClient;
+import com.linkedin.venice.utils.ObjectMapperFactory;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * DvcBlobFinder discovers live DaVinci peer nodes to facilitate blob transfers necessary for bootstrapping the database
+ */
+public class DvcBlobFinder implements BlobFinder {
+  private static final Logger LOGGER = LogManager.getLogger(DvcBlobFinder.class);
+  private static final String TYPE_BLOB_DISCOVERY = "TYPE_BLOB_DISCOVERY";
+  private static final String ERROR_DISCOVERY_MESSAGE =
+      "Error finding DVC peers for blob transfer in store: %s, version: %d, partition: %d";
+  private final TransportClient transportClient;
+  private final String routerUrl;
+
+  public DvcBlobFinder(TransportClient transportClient, String routerUrl) {
```

Comment:
Do we need `routerUrl`? When `TransportClient` is constructed, it's supposed to have the url already. See `HttpTransportClient` as one example

Reply:
Yes, i saw that but need to pass the `routerUrl` separately because `routerUrl` is protected in `HttpTransportClient`.

Reply:
No. My point is since `routerUrl` is already required in some `TransportClient`. Then it's not needed here when sending a request.
https://github.com/linkedin/venice/blob/0ddf4f2e116cd8b14c553b6056abc718c73725a0/clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/transport/HttpTransportClient.java#L149

Reply:
The routerUrl is setting from here 
https://github.com/linkedin/venice/blob/main/clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/ClientFactory.java#L159

and we need to build params -> 
https://github.com/linkedin/venice/pull/1046/files#diff-f7f48dd230ec7f6940dc3dbcd1e2f4334a3a3211fe62f277ee50bc39ad1ebc67R55

maybe we should sync on this as I am not clear on the above explanation